src/Nethermind/Nethermind.Init/Steps/InitializeNetwork.cs

```diff
@@ -29,21 +27,24 @@
 using Nethermind.Network.Rlpx;
 using Nethermind.Network.Rlpx.Handshake;
 using Nethermind.Network.StaticNodes;
-using Nethermind.State.SnapServer;
 using Nethermind.Stats.Model;
 using Nethermind.Synchronization;
 using Nethermind.Synchronization.ParallelSync;
-using Nethermind.Synchronization.SnapSync;
 using Nethermind.Synchronization.Trie;
 using Nethermind.TxPool;
 
 namespace Nethermind.Init.Steps;
 
 public static class NettyMemoryEstimator
 {
-    // Environment.SetEnvironmentVariable("io.netty.allocator.pageSize", "8192");
     private const uint PageSize = 8192;
 
+    public static void SetPageSize()
+    {
+        // For some reason needs to be half page size to get page size
+        Environment.SetEnvironmentVariable("io.netty.allocator.pageSize", (PageSize / 2).ToString((IFormatProvider?)null));
```

Comment:
The reason is the default size set in the netty to 8K https://github.com/Azure/DotNetty/blob/78c5757062a9eb62e2aab2bbb35d3983710f9651/src/DotNetty.Buffers/PooledByteBufferAllocator.cs#L38 and also 4K is the smallest that you can set https://github.com/Azure/DotNetty/blob/78c5757062a9eb62e2aab2bbb35d3983710f9651/src/DotNetty.Buffers/PooledByteBufferAllocator.cs#L33

Why not to set PageSize to 4k literally above.